packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -57,6 +59,63 @@ _ColorsAndStops _interpolateColorsAndStops(
   return _ColorsAndStops(interpolatedColors, interpolatedStops);
 }
 
+/// Base class for transforming gradient shaders without applying the same
+/// transform to the entire canvas.
+///
+/// For example, a [SweepGradient] normally starts its gradation at 3 o'clock
+/// and draws clockwise. To have the sweep appear to start at 6 o'clock, supply
+/// a [GradientRotation] of `0.785398` radians (i.e. `45` degrees).
+@immutable
+abstract class GradientTransform {
+  /// A const constructor that allows subclasses to be const.
+  const GradientTransform();
+
+  /// When a [Gradient] creates its [Shader], it will call this method to
+  /// determine what transform to apply to the shader for the given [Rect] and
+  /// [TextDirection].
+  ///
+  /// Implementers may return null from this method, which achieves the same
+  /// final effect as returning [Matrix4.identity].
+  Matrix4 transform(Rect bounds, {TextDirection textDirection});
+}
+
+/// A [GradientTransform] that rotates the gradient around the center-point of
+/// its bounding box.
+///
+/// For example, the following would rotate a sweep gradient by a quarter turn
+/// clockwise:
+///
+/// ```dart
+/// SweepGradient(
+///   colors: colors,
+///   transform: GradientRotation(0.785398),
+/// );
+/// ```
+@immutable
+class GradientRotation extends GradientTransform {
+  /// Constructs a `GradientRotation` for the specified angle.
+  ///
+  /// The angle is in radians in the clockwise direction.
+  const GradientRotation(this.radians);
+
+  /// The angle of rotation.
```

Comment:
This needs more detail, e.g. direction of rotation (clockwise, counter-clockwise), the units (though it's pretty obvious from the name).